docs/src/main/asciidoc/reactive-postgres-client.adoc

```diff
@@ -0,0 +1,450 @@
+                .thenCompose(r -> client.query("INSERT INTO fruits (name) VALUES ('Orange')"))
+                .thenCompose(r -> client.query("INSERT INTO fruits (name) VALUES ('Pear')"))
+                .thenCompose(r -> client.query("INSERT INTO fruits (name) VALUES ('Apple')"))
+                .toCompletableFuture()
+                .join();
+----
+
+NOTE: Wondering why we need to convert the result to `CompletableFuture` and block until the latest query is completed?
+This code is part of a `@PostConstruct` method and {project-name} invokes it synchronously.
+As a consequence, returning prematurely could lead to serving requests while the database is not ready yet.
+
+That's it!
+So far we have seen how to configure a pooled client and execute simple queries.
+We are now ready to develop the data management code and implement our RESTful endpoint.
+
+== Using
+
+=== Query results traversal
+
+In development mode, the database is set up with a few rows in the `fruits` table.
+To retrieve all the data, we will use the `query` method again:
+
+[source,java]
+----
+CompletionStage<PgRowSet> rowSet = client.query("SELECT id, name FROM fruits ORDER BY name ASC");
+----
+
+When the operation completes, we will get a `PgRowSet` that has all the rows buffered in memory.
+It can be traversed with an `Iterator`:
+
+[source,java]
+----
+CompletionStage<List<Fruit>> fruits = rowSet.thenApply(pgRowSet -> {
+    List<Fruit> list = new ArrayList<>(pgRowSet.size());
+    PgIterator pgIterator = pgRowSet.iterator();
+    while (pgIterator.hasNext()) {
+        list.add(from(pgIterator.next()));
+    }
+    return list;
+});
+----
+
+The `from` method converts a `Row` instance to a `Fruit` instance.
+It is extracted as a convenience for the implementation of the other data management methods:
+
+[source,java]
+.src/main/java/org/acme/vertx/Fruit.java
+----
+private static Fruit from(Row row) {
+    return new Fruit(row.getLong("id"), row.getString("name"));
+}
+----
+
+Putting it all together, the `Fruit.findAll` method looks like:
+
+[source,java]
+.src/main/java/org/acme/vertx/Fruit.java
+----
+    public static CompletionStage<List<Fruit>> findAll(PgPool client) {
+        return client.query("SELECT id, name FROM fruits ORDER BY name ASC").thenApply(pgRowSet -> {
+            List<Fruit> list = new ArrayList<>(pgRowSet.size());
+            PgIterator pgIterator = pgRowSet.iterator();
+            while (pgIterator.hasNext()) {
+                list.add(from(pgIterator.next()));
+            }
+            return list;
+        });
+    }
+----
+
+And the endpoint to get all fruits from the backend:
+
+[source,java]
+.src/main/java/org/acme/vertx/FruitResource.java
+----
+@GET
+public CompletionStage<Response> get() {
+    return Fruit.findAll(client)
+            .thenApply(fruits -> fruits.isEmpty() ? Response.noContent() : Response.ok(fruits))
```

Comment:
Are you sure this is how you write REST APIs for collections? Pretty sure you return an empty collection (JSON/XML) rather than a no-content response, no? How do clients deal with this?

Reply:
This is REST not RPC so I believe no content is better than an empty list. Perhaps you can ask to @patriot1burke :smile: Clients deals with this like then deal with `404` when you search for a single entity that is not found.